File: app/src/main/java/com/rendersoncs/reportform/async/PDFCreateAsync.java
package com.rendersoncs.reportform.async;

import android.content.Context;
import android.os.AsyncTask;
import android.os.Message;
import android.util.Log;

import com.rendersoncs.reportform.itens.ReportItems;
import com.rendersoncs.reportform.util.CreatePDFViewer;

import java.io.File;

import static android.content.ContentValues.TAG;

public class PDFCreateAsync extends AsyncTask<ReportItems, Void, File> {
    private Context context;

    public PDFCreateAsync(Context context){
        this.context = context;
    }

    protected void onPreExecute() {
        super.onPreExecute();
        Log.d(TAG, "onPreExecute call");
    }

    @Override
    protected File doInBackground(ReportItems... reports) {
        Log.d(TAG, "doInBackground AWS call ");

        ReportItems report = reports[0];
        if (report == null) {
            cancel(true);
        }
        File pdfReport = null;
        try {
            pdfReport = new CreatePDFViewer().write(context, report);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return pdfReport;
    }

    protected void onPostExecute(File file) {
        super.onPostExecute(file);
        Log.d(TAG, "onPostExecute call");
    }
}

File: app/src/main/java/com/rendersoncs/reportform/view/services/async/PDFCreateAsync.kt
package com.rendersoncs.reportform.view.services.async

import android.content.ContentValues
import android.content.Context
import android.os.AsyncTask
import android.util.Log
import com.rendersoncs.reportform.itens.ReportItems
import com.rendersoncs.reportform.view.services.util.CreatePDFViewer
import java.io.File

class PDFCreateAsync(private val context: Context) : AsyncTask<ReportItems?, Void?, File?>() {
    override fun onPreExecute() {
        super.onPreExecute()
        Log.d(ContentValues.TAG, "onPreExecute call")
    }

    override fun doInBackground(vararg reports: ReportItems?): File? {
        Log.d(ContentValues.TAG, "doInBackground AWS call ")
        val report = reports[0]
        if (report == null) {
            cancel(true)
        }
        var pdfReport: File? = null
        try {
            pdfReport = CreatePDFViewer().write(context, report)
        } catch (e: Exception) {
            e.printStackTrace()
        }
        return pdfReport
    }

    override fun onPostExecute(file: File?) {
        super.onPostExecute(file)
        Log.d(ContentValues.TAG, "onPostExecute call")
    }

}